packages/merge-on-green/src/merge-logic.ts

```diff
@@ -0,0 +1,537 @@
+    return reviewsPassed;
+  }
+  return reviewsPassed;
+};
+
+mergeOnGreen.merge = async function merge(
+  owner: string,
+  repo: string,
+  pr: number,
+  github: GitHubAPI
+) {
+  const commitInfo = await mergeOnGreen.getPR(owner, repo, pr, github);
+  try {
+    const merge = await github.pulls.merge({
+      owner,
+      repo,
+      pull_number: pr,
+      commit_title: commitInfo.title,
+      commit_message: commitInfo.body,
+      merge_method: 'squash',
+    });
+    return merge;
+  } catch (err) {
+    return null;
+  }
+};
+
+mergeOnGreen.updateBranch = async function updateBranch(
+  owner: string,
+  repo: string,
+  pr: number,
+  github: GitHubAPI
+) {
+  try {
+    const update = await github.pulls.updateBranch({
+      owner,
+      repo,
+      pull_number: pr,
+    });
+    return update;
+  } catch (err) {
+    return null;
+  }
+};
+
+mergeOnGreen.createFailedParam = async function createFailedParam(
+  owner: string,
+  repo: string,
+  pr: number,
+  github: GitHubAPI
+): Promise<CheckParam | null> {
+  const headSha = await mergeOnGreen.getLatestCommit(owner, repo, pr, github);
+  //removed the check for if headSha exists, since we're returning an empty string and the error block will catch it
+  try {
+    const checkParams = github.checks.create({
+      owner,
+      repo,
+      name: 'AutoMerge Failed',
+      head_sha: headSha,
+      status: 'completed',
+      conclusion: 'failure',
+      output: {
+        title: 'Your PR was not mergeable.',
+        summary:
+          'Check your required status checks or requested reviews for failures.',
+        text:
+          'Your PR was not mergeable because either one of your required status checks failed, or one of your required reviews was not approved.' +
+          'Please fix your mistakes, and merge-on-green will run again to attempt to merge it automatically.',
+      },
+    });
+    return checkParams;
+  } catch (err) {
+    return null;
+  }
+};
+
+export async function mergeOnGreen(
+  owner: string,
+  repo: string,
+  pr: number,
+  labelName: string,
+  state: string,
+  github: GitHubAPI
+): Promise<boolean> {
+  console.info(`${owner}/${repo} checking merge on green PR status`);
+  // Checks for reviewers and ensures that the latest review has been
+  // approved.
+  const checkReview = await mergeOnGreen.checkReviews(owner, repo, pr, github);
+  // Checks statuses and check runs, ensuring that the latest version of
+  // a status was a success.
+  const checkStatus = await mergeOnGreen.statusesForRef(
+    owner,
+    repo,
+    pr,
+    labelName,
+    github
+  );
+
+  console.info(
+    `checkReview = ${checkReview} checkStatus = ${checkStatus} state = ${state}`
+  );
+
+  if (checkReview === true && checkStatus === true && state === 'continue') {
+    console.log('Updating branch');
+    await mergeOnGreen.updateBranch(owner, repo, pr, github);
+    console.log('Merging PR');
+    await mergeOnGreen.merge(owner, repo, pr, github);
```

Comment:
let's put a `try`/`catch` around this and `console.log` an error thrown by `mergOnGreen`.